jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -446,84 +378,45 @@ private static void consumeFiles(@NotNull BuildOutputConsumer outputConsumer,
     }
   }
 
-  private static abstract class DirtyFileProcessor<T> implements FileProcessor<ErlangSourceRootDescriptor, ErlangTarget> {
-    private final List<T> myDirtyElements = ContainerUtil.newArrayList();
-
-    public List<T> collectDirtyElements(@NotNull DirtyFilesHolder<ErlangSourceRootDescriptor, ErlangTarget> holder) throws IOException {
-      holder.processDirtyFiles(this);
-      return myDirtyElements;
-    }
-
-    @Override
-    public boolean apply(ErlangTarget erlangTarget,
-                         File file,
-                         ErlangSourceRootDescriptor erlangSourceRootDescriptor) throws IOException {
-      ContainerUtil.addIfNotNull(myDirtyElements, getDirtyElement(erlangTarget, file, erlangSourceRootDescriptor));
-      return true;
+  private static void removeOutputFiles(@NotNull Collection<String> removedFiles, @NotNull File... outputDirectories) {
+    for (File dir : outputDirectories) {
+      List<File> outputErlangModuleFiles = ContainerUtil.concat(getBeams(findErlFiles(removedFiles), dir),
+                                                                getAppsOutput(findAppFiles(removedFiles), dir));
```

Comment:
Place your caret at getBeams and move it forward - methods you call are only used once or twice, yet they're written in no particular order. It's inconvenient to read code like this. Please arrange your methods so that your code can be read sequentially.